### PR TITLE
Add a GA that build GitHub CodeSpaces containers

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

File: .github/workflows/codespace.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  IMAGE_NAME: iog-devx
+  REGISTRY: ghcr.io/input-output-hk
+  DEFAULT_TAG: latest
+
+jobs:
+  build_and_push:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: ['x86_64-linux', 'x86_64-darwin', 'aarch64-linux', 'aarch64-darwin']
```
Comment (hamishmack) [inline .github/workflows/codespace.yml:20]: Will the platforms other than `x86_64-linux` work?

File: .github/workflows/codespace.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        target-platform: ['', '-static', '-windows', '-js']
+        compiler-nix-name: ['ghc961']
```
Comment (hamishmack) [inline .github/workflows/codespace.yml:22]: I think we should be able to use `ghc962` now.

**Review by hamishmack — APPROVED** [review]
It is a little sad we can't build these in nix on hydra and just download them from the cache and upload more or less unmodified (in the GitHub action).

Using the `input-output-hk/actions` script is really nice though, it means the image will match the GitHub actions using that script.

File: .github/workflows/codespace.yml (hunk 1 above)
Comment (yvan-sraka) [reply .github/workflows/codespace.yml:20]: Indeed, no, thanks for having spotted it!

**Review by angerman — APPROVED** [review]
LGTM.

**Pull request merged**